client/client.go

```diff
@@ -516,3 +516,20 @@ func (client *Client) Users() ([]*User, error) {
 	}
 	return result, nil
 }
+
+type debugAction struct {
+	Action string `json:"action"`
+}
+
+// EnsureStateSoon is only useful when writing test code, it will trigger
+// the internal "ensure" loop of the state engine to be run as soon as
+// possible.
+func (client *Client) EnsureStateSoon() error {
```

Comment:
Can we please make this into:

`func (client *Client) Debug(action string, params interface{}, result interface{}) error`

So we're not cooking a static API assuming internal implementation details that will break applications when it gets removed?